### PR TITLE
Update to 2018 edition

### PR DESCRIPTION
**Pull request by torfmaster**
# Summary

This PR updates elf2tab to the 2018 edition. Only the imports have been amended to match new "path clarity" changes.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
-#[macro_use]
-mod util;
 mod cmdline;
 mod header;
-use structopt::StructOpt;
+mod util;
 
 fn main() {
-    let opt = cmdline::Opt::from_args();
+    let opt = Opt::from_args();
```
Comment (bradjc) [inline src/main.rs:16]: Undo

File: src/main.rs [hunk 2]
```diff
@@ -1,24 +1,19 @@
-extern crate chrono;
-extern crate elf;
-extern crate tar;
-#[macro_use]
-extern crate structopt;
-
+use cmdline::Opt;
```
Comment (bradjc) [inline src/main.rs:1]: Using the full path is nice as it tells people right in the source where the very short and not unique "Opt" object is from.

**Comment by torfmaster** [comment]
> Was this intended to be only the first commit? I don't understand the alignment changes. What is wrong with macros?

Sorry, I forgot to comment on that. I tried to add some tests to the crate however, I found out that there is some complexity hidden in the Macros due to hidden conversions (usize to u32 etc.). I wanted to make this explicit using generics but ended up using Numeric traits. Finally, I decided to do this change making the conversions explicit. If found it uncontroversial so I included it. I can split out this change and make it a separate "add tests" PR if you like.

File: src/main.rs (hunk 1 above)
Comment (bradjc) [inline src/main.rs:16]:
```suggestion
    let opt = cmdline::Opt::from_args();
```

File: src/main.rs (hunk 2 above)
Comment (bradjc) [inline src/main.rs:1]:
```suggestion
```

Comment (torfmaster) [reply src/main.rs]: Fixed.

File: src/main.rs (hunk 1 above)
Comment (torfmaster) [reply src/main.rs]: Fixed.

**Comment by torfmaster** [comment]
> Gotcha, that seems reasonable. Is the cmdline change necessary? Otherwise good to go.

Sorry, I didn't get the answer. Shall I split the PR into two?

**Comment by torfmaster** [comment]
I would suggest to close #3 after the tests are merged, although the test coverage could be better... (I hope I set a trend here :)

**Pull request merged**